libs/langchain/langchain/retrievers/kay.py

```diff
@@ -0,0 +1,59 @@
+from __future__ import annotations
+
+from typing import Any, List
+
+from langchain.callbacks.manager import CallbackManagerForRetrieverRun
+from langchain.schema import BaseRetriever, Document
+
+
+class KayAiRetriever(BaseRetriever):
+    """
+    Retriever for Kay.ai datasets.
+
+    To work properly, expects you to have KAY_API_KEY env variable set.
+    You can get one for free at https://kay.ai/.
+    """
+
+    client: Any
+    num_contexts: int
+
+    @classmethod
+    def create(
+        cls,
+        dataset_id: str,
+        data_sources: List[str],
```

Comment:
out of curiosity why not call this data_types (to match the kay arg)?

Reply:
Oh, good catch. We just renamed this arg. Fixed